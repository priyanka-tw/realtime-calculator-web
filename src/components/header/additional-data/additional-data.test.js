import React from 'react';
import {act, fireEvent, render} from '@testing-library/react';
import 'jest-styled-components';
import WS from "jest-websocket-mock";
import AdditionalData from "./additional-data";
import Context from "../../../services/global-context-provider/context";

describe('Additional data', () => {

    let server, client;
    const mockedFn = jest.fn();

    beforeEach(async () => {
        server = new WS("ws://localhost:8080/ws", {jsonProtocol: true});
        client = new WebSocket("ws://localhost:8080/ws");
    });

    afterEach(() => {
        WS.clean();
    });

    it('should render the component', async () => {
        const {container} = render(
            <Context.Provider value={{
                username: 'test-username',
                socket: client,
                setUsername: mockedFn
            }}>
                <AdditionalData/>
            </Context.Provider>);
        await server.connected;
        server.send('message');

        expect(container).toMatchSnapshot();
    });

    it('should handle logout on click of logout button', async () => {
        const {getByTestId} = render(
            <Context.Provider value={{
                username: 'test-username',
                socket: client,
                setUsername: mockedFn
            }}>
                <AdditionalData/>
            </Context.Provider>);
        await server.connected;

        await act(async () => {
            await fireEvent.click(getByTestId('logout-button'));
        });

        await expect(server).toReceiveMessage({"data": "test-username", "event": "logout"});
        expect(mockedFn).toHaveBeenCalledWith('');
    });

    it('should update logged in users data according to the message received', async () => {
        const {container} = render(
            <Context.Provider value={{
                username: 'test-username',
                socket: client,
                setUsername: mockedFn
            }}>
                <AdditionalData/>
            </Context.Provider>);
        await server.connected;
        server.send({event: 'logged in users', data: '5'});

        expect(container).toMatchSnapshot();
    });
});
