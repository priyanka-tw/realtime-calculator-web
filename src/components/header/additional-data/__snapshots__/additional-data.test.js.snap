// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Additional data should render the component 1`] = `
.c1 {
  font-size: 14px;
  font-family: verdana;
  margin-left: 400px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 100%;
}

<div>
  <div
    class="c0"
  >
    <div>
      <h3
        class="c1"
        data-testid="greet-message"
      >
        Hi 
        Test-username
        , currently connected users: 
        1
      </h3>
    </div>
    <button
      data-testid="logout-button"
      type="button"
    >
       logout
    </button>
  </div>
</div>
`;

exports[`Additional data should update logged in users data according to the message received 1`] = `
.c1 {
  font-size: 14px;
  font-family: verdana;
  margin-left: 400px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 100%;
}

<div>
  <div
    class="c0"
  >
    <div>
      <h3
        class="c1"
        data-testid="greet-message"
      >
        Hi 
        Test-username
        , currently connected users: 
        5
      </h3>
    </div>
    <button
      data-testid="logout-button"
      type="button"
    >
       logout
    </button>
  </div>
</div>
`;
